desc/io/efit_io.py

```diff
@@ -0,0 +1,234 @@
+"""Utilities for reading EFIT geqdsk files."""
+
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy.interpolate import CubicSpline
+
+
+def efit_to_desc(g, M, N=0, L=None, sep_dev=0):
+    """Create a DESC equilibrium from an EFIT geqdsk file.
+
+    The equilibrium will not be solved, this just creates an equilibirum
+    with the correct boundary and profiles.
+
+    Parameters
+    ----------
+    g : path-like or dict
+        Path to geqdsk file, or dict of already read parameters.
+    M : int
+        Poloidal resolution of the equilibrium. Used for fitting the boundary.
+    N : int, optional
+        Toroidal resolution of the equilibrium.
+    L : int, optional
+        Radial resolution of the equilibrium. Used for fitting profiles.
+        If None, spline profiles will be used. Otherwise they are fit
+        to even power series in rho.
+    sep_dev : float
+        Deviation from separatrix. If sep_dev == 0, then the boundary
+         is read from the file.sep_dev = 1 corresponds to the magnetic axis.
+         Typical values should be 0.0 < sep_dev < 0.05
+
+    Returns
+    -------
+    eq : Equilibrium
+        DESC equilibrium with boundary shape, pressure, iota, and total toroidal flux
+        from geqdsk file.
+    """
+    from desc.basis import DoubleFourierSeries
+    from desc.equilibrium import Equilibrium
+    from desc.geometry import FourierRZToroidalSurface
+    from desc.grid import LinearGrid
+    from desc.profiles import SplineProfile
+    from desc.transform import Transform
+
+    if not isinstance(g, dict):
+        g = read_gfile(g, sep_dev=sep_dev)
+    ns = len(g["q"])
+    chiN = np.linspace(0, 1, ns)
+    chi_scale = g["chi_boundary"] - g["chi_axis"]
+    # chi = poloidal flux
+    # psi = toroidal flux
+
+    # iota = dchi / dpsi = dchi / dchiN * dchiN / dpsi = dchiN / dpsi * (chi_scale)
+    # dpsi = 1/iota dchi = q dchiN
+    # psi = integral(q dchiN) * (chi_scale)
+
+    # q * (chi(1) - chi(0)) as function of chiN
+    q_spline = CubicSpline(chiN, g["q"] * chi_scale)
+    # this gives psi(chiN)
+    psi_spline = q_spline.antiderivative()
+
+    psi = psi_spline(chiN)
+    Psi_b = psi[-1]  # boundary flux
+    psiN = psi / Psi_b
+    rho = np.sqrt(psiN)
+
+    pressure = SplineProfile(g["pres"], rho)
+    iota = SplineProfile(1 / g["q"], rho)
```

Comment:
is `1/q` the same as iotabar = $/iota/ 2\pi$ or is it just $/iota$?

Reply:
we use iotabar

Reply:
It should just be iota.